# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: intl/locale/LocaleService.cpp
# Commit: a345e05df151
# Full Hash: a345e05df151563625fe05ec63094b3dae2c9379
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-07 21:31:11
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 288a67aed661 -r a345e05df151 intl/locale/LocaleService.cpp
--- a/intl/locale/LocaleService.cpp	Tue Jul 07 04:35:08 2020 +0000
+++ b/intl/locale/LocaleService.cpp	Tue Jul 07 04:36:27 2020 +0000
@@ -295,13 +295,14 @@
 
 static bool GetGREFileContents(const char* aFilePath, nsCString* aOutString) {
   // Look for the requested file in omnijar.
-  RefPtr<nsZipArchive> zip = Omnijar::GetReader(Omnijar::GRE);
+  RefPtr<CacheAwareZipReader> zip = Omnijar::GetReader(Omnijar::GRE);
   if (zip) {
-    nsZipItemPtr<char> item(zip, aFilePath);
-    if (!item) {
+    uint32_t length;
+    const uint8_t* data = zip->GetData(aFilePath, &length);
+    if (!data) {
       return false;
     }
-    aOutString->Assign(item.Buffer(), item.Length());
+    aOutString->Assign(reinterpret_cast<const char*>(data), length);
     return true;
   }
 